sdk/lib/core/uri.dart

```diff
@@ -24,24 +24,52 @@ const int _LOWER_CASE_Z = 0x7A;
 
 const String _hexDigits = "0123456789ABCDEF";
 
-/// A parsed URI, such as a URL.
+/// A parsed URI, such as a URL. URI is Uniform Resource Identifier. URL is
+/// Uniform Resource Locator, a.k.a. a web address.
 ///
-/// To create a URI with specific components, use [new Uri]:
+/// Use this class to represent web addresses (`https://...`), local file
+/// locations (`C:\file` on Windows or `/file` on Unixes), or local folder
+/// locations (`C:\folder\` or `/folder/`).
+///
+/// You can create web URIs either by specifying the scheme (protocol) as a 
+/// parameter (`Uri(scheme: 'https', ...)`) or using the optional [Uri.https]
+/// or [Uri.http] constructor.
+///
+/// To create a URI, use [new Uri]:
 /// ```dart
-/// var httpsUri = Uri(
+/// // Use a final if the Uri contents never change
+/// final dartLibTour = Uri(
 ///     scheme: 'https',
 ///     host: 'dart.dev',
 ///     path: '/guides/libraries/library-tour',
 ///     fragment: 'numbers');
-/// print(httpsUri); // https://dart.dev/guides/libraries/library-tour#numbers
+/// print(dartLibTour); // https://dart.dev/guides/libraries/library-tour#numbers
+/// ```
 ///
-/// httpsUri = Uri(
+/// The optional parts of the URL at the end come in two flavors: they can
+/// be fragments or query parameters. For convenience, the query Params
+/// can be provided in the form of a query string, or piecemeal in the form
+/// of a map of query parameter names and their values.
+///
+/// For example, using `queryParameters` instead of the `query` `String?`:
+///
+/// ```dart
+/// // Use a var if you want to change the Uri later
+/// var exampleUri = Uri(scheme: 'https', host: 'dart.dev');
+/// print(exampleUri); // https://dart.dev
+/// // uh oh, that's not what we wanted...let's change it:
+/// // https://en.wikipedia.org/w/index.php?title=Uniform_Resource_Identifier&action=history
+/// exampleUri = Uri(
 ///     scheme: 'https',
-///     host: 'example.com',
-///     path: '/page/',
-///     queryParameters: {'search': 'blue', 'limit': '10'});
-/// print(httpsUri); // https://example.com/page/?search=blue&limit=10
+///     host: 'wikipedia.org',
+///     path: '/w/index.php',
+///     queryParameters: {'title': 'Uniform_Resource_Identifier', 'action':'history'});
+/// print(exampleUri); // https://en.wikipedia.org/w/index.php?title=Uniform_Resource_Identifier&action=history
+/// ```
 ///
+/// Make a mailto link to email someone in the user's default Mail application:
```

Comment:
Don't think mentioning a "default Mail application" is useful here. Maybe as an example, but that still presumes that the code runs in a setting where there is a default mail application.